nrf_security/src/legacy/CMakeLists.txt

```diff
@@ -46,6 +46,13 @@ if(CONFIG_MBEDTLS_LEGACY_CRYPTO_C OR
   )
 endif()
 
+if(CONFIG_MBEDTLS_PSA_CRYPTO_SPM AND CONFIG_MBEDTLS_HKDF_C)
```

Comment:
Complete this with || MBEDTLS_PSA_BUILTIN_ALG_HKDF so we have a way to remove CONFIG_MBEDTLS_HKDF_C more cleanly when legacy configuration is dropped?

Reply:
Ah. we aren't using the BUILTIN config, but rather relying on legacy configuration for the recursion problem. Resolving myself